nvflare/lighter/provision.py

```
@@ -94,55 +95,81 @@ def handle_provision(args):
         raise ValueError(f"API version expected 3 but found {api_version}")
-
     project_name = project_dict.get("name")
     project_description = project_dict.get("description", "")
     participants = list()
     for p in project_dict.get("participants"):
         participants.append(Participant(**p))
-    if args.add_user:
-        try:
-            extra = load_yaml(os.path.join(current_path, args.add_user))
-            extra.update({"type": "admin"})
-            participants.append(Participant(**extra))
-        except BaseException:
-            print("** Error during adding user **")
-            print("The yaml file format is")
-            print(adding_user_error_msg)
-            exit(0)
-    if args.add_client:
-        try:
-            extra = load_yaml(os.path.join(current_path, args.add_client))
-            extra.update({"type": "client"})
-            participants.append(Participant(**extra))
-        except BaseException as e:
-            print("** Error during adding client **")
-            print("The yaml file format is")
-            print(adding_client_error_msg)
-            exit(0)
-
+    if add_user_file_path:
+        add_extra_users(add_user_file_path, participants)
+    if add_client_file_path:
+        add_extra_clients(add_client_file_path, participants)
     project = Project(name=project_name, description=project_description, participants=participants)
-
     n_servers = len(project.get_participants_by_type("server", first_only=False))
     if n_servers > 2:
-        print(
-            f"Configuration error: Expect 2 or 1 server to be provisioned.  {project_full_path} contains {n_servers} servers."
-        )
-        return
-
-    builders = list()
-    for b in project_dict.get("builders"):
-        path = b.get("path")
-        args = b.get("args")
-        builders.append(instantiate_class(path, args))
+        raise (f"Configuration error: Expect 2 or 1 server to be provisioned. project contains {n_servers} servers.")
```

should raise an Exception?